### PR TITLE
Only Admin can set or change the clustertemplate.enforced field

### PR DESCRIPTION
**Pull request by prachidamle**
https://github.com/rancher/rancher/issues/21200

API validation to ensure ClusterTemplate can be enforced only by an Admin

### CODE REVIEW

File: pkg/api/store/clustertemplate/store.go [hunk 1]
```diff
@@ -9,26 +9,40 @@ import (
...
+		Users:     users,
+		GrbLister: grbLister,
+		GrLister:  grLister,
 	}
 	return storeWrapped
 }
 
 type Store struct {
 	types.Store
+	Users     v3.UserInterface
```
Comment (mrajashree) [inline pkg/api/store/clustertemplate/store.go]: Maybe we don't need to export these?

File: pkg/api/store/clustertemplate/store.go [hunk 2]
```diff
@@ -157,3 +177,39 @@ func isDefaultTemplateRevision(apiContext *types.APIContext, id string) (bool, e
...
+		}
+	}
+	if enforcedFlagChanged {
+		//only admin can set the flag
+		ma := gaccess.MemberAccess{
+			Users:     p.Users,
+			GrLister:  p.GrLister,
+			GrbLister: p.GrbLister,
+		}
+		//if user is admin, no checks needed
```
Comment (mrajashree) [inline pkg/api/store/clustertemplate/store.go]: This comment is misplaced maybe? :)

File: pkg/api/store/clustertemplate/store.go [hunk 3]
```diff
@@ -157,3 +177,39 @@ func isDefaultTemplateRevision(apiContext *types.APIContext, id string) (bool, e
...
+		callerID := apiContext.Request.Header.Get(gaccess.ImpersonateUserHeader)
+		isAdmin, err := ma.IsAdmin(callerID)
+		if err != nil {
+			return err
+		}
+		if !isAdmin {
+			return httperror.NewAPIError(httperror.PermissionDenied, fmt.Sprintf("ClusterTemplate's %v field cannot be changed", managementv3.ClusterTemplateFieldEnforced))
```
Comment (mrajashree) [inline pkg/api/store/clustertemplate/store.go]: Can we change the error message to show that non-admin user made this request hence enforcement can't be set?

Comment (prachidamle) [reply pkg/api/store/clustertemplate/store.go]: Yeah, deliberately keeping the error message generic, reason is to not reveal any unnecessary rbac information.

**Pull request merged**